src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java

```diff
@@ -81,4 +81,52 @@ public void testGetLastTradedPriceEmpty() throws Exception {
         .andExpect(jsonPath("$", is(Collections.emptyList())))
         .andReturn();
   }
+
+  @Test
+  public void testGetHistoricalPriceForRange() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=FB&range=5d")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].symbol", is("FB")))
+        .andExpect(jsonPath("$[0].open").value(new BigDecimal("301.945")))
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("301.47")))
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("301.98")))
+        .andExpect(jsonPath("$[0].low").value(new BigDecimal("297.39")))
+        .andExpect(jsonPath("$[0].volume", is(14863474)))
+        .andExpect(jsonPath("$[0].date", is("2021-04-21")))
+        .andReturn();
```

Comment:
Can you add an assertion to confirm the number of response objects match the number of days you queried for and maybe confirm the last response (day 5) has the values you are expecting? Looks like we're just asserting on the first response.

Reply:
Added the condition in the latest commit. Getting a response of length 3 because the api takes in variables by calendar days but responds in business days. So for 5d, we are getting a response of 3 business day historical prices